frontend/src/metabase/admin/settings/app/components/SettingsEditor/SettingsEditor.jsx

```diff
@@ -81,18 +81,28 @@ class SettingsEditor extends Component {
   }
 
   /**
-   * @param {Object} setting
+   * @param {Object|import("metabase-types/api").SettingKey} settingOrKey
    * @param {*} newValue
    * @param {Object} options - allows external callers in setting's that user custom components to hook into the success or failure of the update
    * @param {function} [options.onChanged] - callback fired after the setting has been updated
    * @param {function} [options.onError] - callback fired after the setting has failed to update
    */
-  updateSetting = async (setting, newValue, options) => {
-    const { settingValues, updateSetting, reloadSettings, dispatch } =
-      this.props;
+  updateSetting = async (settingOrKey, newValue, options) => {
```

Comment:
Modify the API to allow using this updater without having to pass the whole setting which is more cumbersome.

Reply:
I agree this pattern is more cumbersome, we're working very hard to address it in another way.